Experimental/MultiplayerUseCases/Assets/Common/Scripts/NetworkManagerUI.cs

```diff
@@ -1,71 +1,102 @@
+        Button m_QuitSceneButton;
+
         [SerializeField] string m_SelectionScreenSceneName;
         [SerializeField] GameObject m_ServerOnlyOverlay;
-        Button[] m_Buttons;
 
         void Awake()
         {
-            m_ServerButton.onClick.AddListener(StartServer);
-            m_HostButton.onClick.AddListener(StartHost);
-            m_ClientButton.onClick.AddListener(StartClient);
-            m_DisconnectButton.onClick.AddListener(Disconnect);
-            m_QuitSceneButton.onClick.AddListener(QuitScene);
-            m_Buttons = new Button[] { m_ServerButton, m_HostButton, m_ClientButton };
+            var uiDocument = GetComponent<UIDocument>();
+            m_Root = uiDocument.rootVisualElement;
+
+            m_ServerVisualHighlight = m_Root.Q<VisualElement>("ServerVisualHighlight");
+            m_HostVisualHighlight = m_Root.Q<VisualElement>("HostVisualHighlight");
+            m_ClientVisualHighlight = m_Root.Q<VisualElement>("ClientVisualHighlight");
+            m_AddressInputField = m_Root.Q<TextField>("IPAddressField");
+            m_AddressInputField.SetValueWithoutNotify(k_DefaultIP);
+            m_PortInputField = m_Root.Q<TextField>("PortField");
+            m_PortInputField.SetValueWithoutNotify(k_DefaultPort.ToString());
+
+            m_ServerButton = UIElementsUtils.SetupButton("ServerButton", StartServer, true, m_Root, "Server", "Starts the Server");
+            m_HostButton = UIElementsUtils.SetupButton("HostButton", StartHost, true, m_Root, "Host", "Starts the Host");
+            m_ClientButton = UIElementsUtils.SetupButton("ClientButton", StartClient, true, m_Root, "Client", "Starts the Client");
+            m_DisconnectButton = UIElementsUtils.SetupButton("DisconnectButton", Disconnect, false, m_Root, "Disconnect", "Disconnects participant from session");
+            UIElementsUtils.SetupButton("QuitSceneButton", QuitScene, true, m_Root, "Quit Scene", "Quits scene and brings you back to the scene selection screen");
         }
 
         void Start()
         {
-            SetButtonStateAndColor(m_DisconnectButton, false, false);
             m_ServerOnlyOverlay.gameObject.SetActive(false);
         }
 
         void StartServer()
         {
+            SetNetworkPortAndAddress(ushort.Parse(m_PortInputField.value), m_AddressInputField.value, k_DefaultServerListenAddress);
+            NetworkManager.Singleton.GetComponent<UnityTransport>().ConnectionData.ServerListenAddress = m_AddressInputField.value;
             NetworkManager.Singleton.StartServer();
-            EnableAndHighlightButtons(m_ServerButton, false);
-            SetButtonStateAndColor(m_DisconnectButton, false, true);
             m_ServerOnlyOverlay.gameObject.SetActive(true);
+
+            Button[] buttons = new Button[] {m_ClientButton, m_HostButton};
+            EnableButtons(buttons, false);
+            m_DisconnectButton.SetEnabled(true);
+            m_ServerVisualHighlight.visible = true;
```

Comment:
1. This code is repeated across the Start<stuff> methods, which is not good. Should be an external method that each one of those calls
2. As mentioned [here](https://unity.slack.com/archives/C07371XNH62/p1718635715221839?thread_ts=1718630212.048659&cid=C07371XNH62), the server button should be disabled as well. You don't want people to play "Server" again while they're in server mode. (same goes for any other mode). The on
3. The overlay is what tells you that you are in dedicated server mode, you don't even need the highlight.